tests/devices/test_local_channel.py

```diff
@@ -147,9 +147,9 @@ async def test_message_decode_error(local_channel: LocalChannel, caplog: pytest.
     local_channel._data_received(b"invalid_payload")
     await asyncio.sleep(0.01)  # yield
 
-    assert len(caplog.records) == 1
-    assert caplog.records[0].levelname == "WARNING"
-    assert "Failed to decode message" in caplog.records[0].message
+    warning_records = [record for record in caplog.records if record.levelname == "WARNING"]
```

Comment:
I think theres something you can do like `with caplog.at_level(logging.WARNING):` as well surrounding this, here and below